batch/batch/client.py

```diff
@@ -211,3 +214,28 @@ def create_job(self,
     def create_batch(self, attributes=None):
         batch = self.api.create_batch(self.url, attributes)
         return Batch(self, batch['id'])
+
+    job_yaml_schema = {
+        'spec': schemas.pod_spec,
+        'type': {'type': 'string', 'allowed': ['execute']},
+        'name': {'type': 'string'},
+        'dependsOn': {'type': 'list', 'schema': {'type': 'string'}},
+    }
+    job_yaml_validator = cerberus.Validator(job_yaml_schema)
+
+    def create_batch_from_file(self, file):
+        job_ids = {}
+        batch = self.create_batch()
+        for doc in yaml.load(file):
+            if not BatchClient.job_yaml_validator.validate(doc):
+                raise BatchClient.job_yaml_validator.errors
+            spec = doc['spec']
+            type = doc['type']
+            name = doc['name']
+            dependsOn = doc.get('dependsOn', [])
+            if type == 'execute':
+                job = batch.create_job(
+                    parent_ids=[job_ids[x] for x in dependsOn],
```

Comment:
Should we have a nicer error message if a 'dependsOn' field isn't a previously seen job name?